### PR TITLE
Issues/72

### PR DESCRIPTION
**Pull request by fjaviersanchez**
Add ability to use AND/OR degrade

### CODE REVIEW

**Comment by erykoff** [comment]
Quick comment -- there needs to be support/tests for and/or with straight integer maps, even if they aren't wide masks.

File: tests/test_degrade.py [hunk 1]
```diff
@@ -111,60 +112,122 @@ def test_degrade_map_recarray(self):
...
+        Test HealSparse.degrade AND functionality with WIDE_MASK
         """
-        
+
         nside_coverage = 32
         nside_map = 256
         nside_map2 = 64
         sparse_map = healsparse.HealSparseMap.make_empty(nside_coverage, nside_map,
-                                                 WIDE_MASK, wide_mask_maxbits=7)
+                                                         WIDE_MASK, wide_mask_maxbits=7)
```
Comment (erykoff) [inline tests/test_degrade.py:176]: The and/or wide_mask tests need two separate tests each, unfortunately, one of them with < 8 bits (like this) and another with > 8 bits.  The one with > 8 bits needs to ensure that some pixels have a low bit set, some pixels have a high bit set, and some have both.  (I have learned this the hard way!)

**Review by erykoff — CHANGES_REQUESTED** [review]
See comments about additional wide_mask tests.  Otherwise looks good.

File: tests/test_degrade.py (hunk 1 above)
Comment (fjaviersanchez) [reply tests/test_degrade.py:176]: Great suggestion, this allowed me to catch a couple of bugs.

**Review by erykoff — APPROVED** [review]
Looks good, I think the tests have good coverage!

**Pull request merged**